From 703b1dc1376613ae9590c76045320053e81038b1 Mon Sep 17 00:00:00 2001
From: Sooyong Suk <s.suk@samsung.corp-partner.google.com>
Date: Fri, 22 May 2026 18:45:51 +0900
Subject: [PATCH 205/223] ANDROID: GKI: update symbol list for galaxy

2 function symbol(s) added
  'int __traceiter_android_rvh_do_swap_page_relax(void*, swp_entry_t, bool*)'
  'int __traceiter_android_vh_do_swap_page_done(void*, swp_entry_t)'

2 variable symbol(s) added
  'struct tracepoint __tracepoint_android_rvh_do_swap_page_relax'
  'struct tracepoint __tracepoint_android_vh_do_swap_page_done'

Bug: 514180992
Bug: 521135041
Change-Id: Ic87ed6decf9051d8f27cd6036c276061611e2508
(cherry picked from commit d7c970e477a170f747392effd61753561f1e7251)
Signed-off-by: Sooyong Suk <s.suk@samsung.corp-partner.google.com>
---
 gki/aarch64/abi.stg        | 47 ++++++++++++++++++++++++++++++++++++++
 gki/aarch64/symbols/galaxy |  4 ++++
 2 files changed, 51 insertions(+)

diff --git a/gki/aarch64/abi.stg b/gki/aarch64/abi.stg
index 09836cefc486..56295efbe0dd 100644
--- a/gki/aarch64/abi.stg
+++ b/gki/aarch64/abi.stg
@@ -416669,6 +416669,13 @@ function {
   parameter_id: 0x4585663f
   parameter_id: 0x1bf16028
 }
+function {
+  id: 0x9b379132
+  return_type_id: 0x6720d32f
+  parameter_id: 0x18bd6530
+  parameter_id: 0x27162aac
+  parameter_id: 0x11cfee5a
+}
 function {
   id: 0x9b37ef14
   return_type_id: 0x6720d32f
@@ -440525,6 +440532,15 @@ elf_symbol {
   type_id: 0x9b6eb620
   full_name: "__traceiter_android_rvh_do_sp_pc_abort"
 }
+elf_symbol {
+  id: 0x8d7abadb
+  name: "__traceiter_android_rvh_do_swap_page_relax"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xf6d205b6
+  type_id: 0x9b379132
+  full_name: "__traceiter_android_rvh_do_swap_page_relax"
+}
 elf_symbol {
   id: 0x255d03b3
   name: "__traceiter_android_rvh_dpm_prepare"
@@ -443054,6 +443070,15 @@ elf_symbol {
   type_id: 0x9b2612dc
   full_name: "__traceiter_android_vh_do_swap_page"
 }
+elf_symbol {
+  id: 0xa1d54c92
+  name: "__traceiter_android_vh_do_swap_page_done"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xba7470ba
+  type_id: 0x9b33e2c9
+  full_name: "__traceiter_android_vh_do_swap_page_done"
+}
 elf_symbol {
   id: 0x02ccf9e7
   name: "__traceiter_android_vh_do_sync_mmap_readahead"
@@ -448571,6 +448596,15 @@ elf_symbol {
   type_id: 0x18ccbd2c
   full_name: "__tracepoint_android_rvh_do_sp_pc_abort"
 }
+elf_symbol {
+  id: 0x17ec58c1
+  name: "__tracepoint_android_rvh_do_swap_page_relax"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_rvh_do_swap_page_relax"
+}
 elf_symbol {
   id: 0x6de60f21
   name: "__tracepoint_android_rvh_dpm_prepare"
@@ -451100,6 +451134,15 @@ elf_symbol {
   type_id: 0x18ccbd2c
   full_name: "__tracepoint_android_vh_do_swap_page"
 }
+elf_symbol {
+  id: 0xfcc0a874
+  name: "__tracepoint_android_vh_do_swap_page_done"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0x7c5aa8a7
+  type_id: 0x18ccbd2c
+  full_name: "__tracepoint_android_vh_do_swap_page_done"
+}
 elf_symbol {
   id: 0x1b92a821
   name: "__tracepoint_android_vh_do_sync_mmap_readahead"
@@ -528165,6 +528208,7 @@ interface {
   symbol_id: 0x81449840
   symbol_id: 0x611114e0
   symbol_id: 0x76eb11ee
+  symbol_id: 0x8d7abadb
   symbol_id: 0x255d03b3
   symbol_id: 0x6682daab
   symbol_id: 0x24dfaeb0
@@ -528446,6 +528490,7 @@ interface {
   symbol_id: 0xb12728da
   symbol_id: 0x20f34a30
   symbol_id: 0x54bc5972
+  symbol_id: 0xa1d54c92
   symbol_id: 0x02ccf9e7
   symbol_id: 0x9dbd7b92
   symbol_id: 0x2576f1c7
@@ -529059,6 +529104,7 @@ interface {
   symbol_id: 0x411e7a56
   symbol_id: 0xcfcbb846
   symbol_id: 0x1d7748c4
+  symbol_id: 0x17ec58c1
   symbol_id: 0x6de60f21
   symbol_id: 0xf9b36d15
   symbol_id: 0xbade8646
@@ -529340,6 +529386,7 @@ interface {
   symbol_id: 0x474d211c
   symbol_id: 0xe66acad6
   symbol_id: 0xeb9f1c78
+  symbol_id: 0xfcc0a874
   symbol_id: 0x1b92a821
   symbol_id: 0xe2d7542c
   symbol_id: 0x15374b6d
diff --git a/gki/aarch64/symbols/galaxy b/gki/aarch64/symbols/galaxy
index 717e4dc5b3ce..e4fbbdc63463 100644
--- a/gki/aarch64/symbols/galaxy
+++ b/gki/aarch64/symbols/galaxy
@@ -53,6 +53,7 @@
   __traceiter_android_rvh_do_el1_undef
   __traceiter_android_rvh_do_sea
   __traceiter_android_rvh_do_sp_pc_abort
+  __traceiter_android_rvh_do_swap_page_relax
   __traceiter_android_rvh_handle_bad_stack
   __traceiter_android_rvh_panic_unhandled
   __traceiter_android_rvh_process_madvise_bypass
@@ -68,6 +69,7 @@
   __traceiter_android_vh_dma_heap_buffer_alloc_start
   __traceiter_android_vh_dma_heap_buffer_alloc_end
   __traceiter_android_vh_do_read_fault
+  __traceiter_android_vh_do_swap_page_done
   __traceiter_android_vh_dpm_wait_finish
   __traceiter_android_vh_dpm_wait_start
   __traceiter_android_vh_filemap_fault_start
@@ -141,6 +143,7 @@
   __tracepoint_android_rvh_do_el1_undef
   __tracepoint_android_rvh_do_sea
   __tracepoint_android_rvh_do_sp_pc_abort
+  __tracepoint_android_rvh_do_swap_page_relax
   __tracepoint_android_rvh_handle_bad_stack
   __tracepoint_android_rvh_panic_unhandled
   __tracepoint_android_rvh_process_madvise_bypass
@@ -156,6 +159,7 @@
   __tracepoint_android_vh_dma_heap_buffer_alloc_start
   __tracepoint_android_vh_dma_heap_buffer_alloc_end
   __tracepoint_android_vh_do_read_fault
+  __tracepoint_android_vh_do_swap_page_done
   __tracepoint_android_vh_dpm_wait_finish
   __tracepoint_android_vh_dpm_wait_start
   __tracepoint_android_vh_filemap_fault_start
-- 
2.50.1

